core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -114,15 +114,6 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
 
   def shuffleMergeAllowed : Boolean = _shuffleMergeAllowed
 
-  // By default, shuffle merge is enabled for ShuffleDependency if shuffleMergeAllowed
-  private[this] var _shuffleMergeEnabled = shuffleMergeAllowed
-
-  private[spark] def setShuffleMergeEnabled(shuffleMergeEnabled: Boolean): Unit = {
-    _shuffleMergeEnabled = shuffleMergeEnabled
-  }
-
-  def shuffleMergeEnabled : Boolean = _shuffleMergeEnabled
```

Comment:
Let us keep this method - and make it
`def shuffleMergeEnabled = mergerLocs.nonEmpty`
What I meant was, let us remove the boolean `_shuffleMergeEnabled`

So that rest of the code can depend on this as a state and not need to inspect the mergerLocs directly

Reply:
Yeah. thats true. We are using `mergerLocs.isEmpty` in other places as well. If we use `shuffleMergeEnabled (mergerLocs.nonEmpty)` and `mergerLocs.isEmpty|nonEmpty` in few other places, wouldn't it be confusing? Do you suggest to replace all the usages of `mergerLocs.isEmpty|nonEmpty` appropriately with `shuffleMergeEnabled | !shuffleMergeEnabled`?

Reply:
In that case, should we also make `shuffleMergeAllowed` as well `private[spark]`. For now it is better to make everything `private` related to push-based shuffle right? Thoughts?

Reply:
> We are using mergerLocs.isEmpty in other places as well. If we use shuffleMergeEnabled (mergerLocs.nonEmpty) and mergerLocs.isEmpty|nonEmpty in few other places, wouldn't it be confusing?

Yes, we should not blindly replace - only when the intent is to check if shuffle merge is enabled; though I would expect a empty/nonEmpty check against mergerLocs to pretty much be a proxy for is shuffle merge enabled.
For example, the changes in the tests - the earlier intent was to check if merge was enabled; which got replaced with checks against mergerLocs

Reply:
> In that case, should we also make shuffleMergeAllowed as well private[spark]. For now it is better to make everything private[spark] related to push-based shuffle right? Thoughts?

`shuffleMergeAllowed` is similar to `shuffleMergeEnabled` - and has value as a developer api, which developers can query for - the setters dont.
FinalizeTask on other hand is a purely internal impl detail.

Reply:
@mridulm How about `incPushCompleted` and `newShuffleMergeState`? Those should also be `private[spark]` right given that it is internal impl detail?

Reply:
Good call on `incPushCompleted`.
Both `setMergerLocs` and `newShuffleMergeState` are in released versions, which makes it more difficult to make the change.

Reply:
+CC @tgravescs, @Ngone51

Reply:
@venkata91 Let us fix the changes from master, which are not in 3.2/older - and open a jira for the others as a follow up.

Reply:
Created a separate JIRA to address this issue - https://issues.apache.org/jira/browse/SPARK-38064